MobileNav/SDL/SDL UI/Map/MapManager.swift

```diff
@@ -0,0 +1,130 @@
+        mapView.setCenter(coordinate, zoomLevel: 12, animated: false)
+
+        newMapCenterPoint = mapView.center
+        mapZoomLevel = mapView.zoomLevel
+
+        if NotificationQueue.shared.lastNotification != nil {
+            NotificationCenter.default.post(name: .centerMapOnPlace, object: nil)
+        }
+    }
+
+    func updateScreen() {
+        let mapCenterPoint = self.mapViewCenterPoint
+        let newMapCenterPoint = self.newMapCenterPoint
+        guard mapCenterPoint != newMapCenterPoint else { return }
+        
+        
+        // Set new center
+        let newMapCenterCoordinate = self.mapView.convert(newMapCenterPoint, toCoordinateFrom: self.mapView)
+        self.mapView.setCenter(newMapCenterCoordinate, animated: false)
+        self.newMapCenterPoint = self.mapView.center
+        self.mapView.zoomLevel = self.mapZoomLevel
+    }
+
+    func zoomIn() {
+        mapView.setZoomLevel(mapView.zoomLevel + 1, animated: true)
+    }
+
+    func zoomOut() {
+        mapView.setZoomLevel(mapView.zoomLevel - 1, animated: true)
+    }
+
+    func centerLocation(lat:CLLocationDegrees, long:CLLocationDegrees) {
+        self.mapView.setCenter(CLLocationCoordinate2DMake(lat, long), animated: false)
+    }
+
+}
+
+extension MapManager {
+    var mapManagerTouchHandler: TouchHandler? {
+        return { [unowned self] (touchPoint: CGPoint, scale: CGFloat?, touchType: TouchType) in
+            switch touchType {
+            case .singleTap:
+                self.zoom(to: touchPoint)
+            case .doubleTap:
+                self.zoom(out: touchPoint)
+            case .panMoved:
+                self.panMoved(displacement: touchPoint)
+            case .pinchMoved:
+                self.pinchMoved(centerPoint: touchPoint, scale: scale ?? 1.0)
+            case .panStarted, .pinchStarted: break
+            case .panEnded, .pinchEnded: break
+            }
+            self.updateScreen()
+        }
+    }
+
+    /// Zooms-in and centers around the tap gesture point.
+    ///
+    /// - Parameters:
+    ///   - touchPoint: The location of the tap on the screen
+    private func zoom(to touchPoint: CGPoint) {
+        newMapCenterPoint = touchPoint
+        let newZoomLevel = mapView.zoomLevel + 1
+        mapZoomLevel = newZoomLevel
+    }
+
+    /// Zooms-out and centers around the tap gesture point.
+    ///
+    /// - Parameters:
+    ///   - touchPoint: The location of the tap on the screen
+    private func zoom(out touchPoint: CGPoint) {
+        newMapCenterPoint = touchPoint
+        let newZoomLevel = mapView.zoomLevel - 1
+        mapZoomLevel = newZoomLevel
+    }
+
+    /// Pans the map.
+    ///
+    /// - Parameter displacement: The distance the of the finger drag
+    private func panMoved(displacement: CGPoint) {
+        newMapCenterPoint = CGPoint(
+            x: newMapCenterPoint.x + displacement.x,
```

Comment:
When I use my finger to pan horizontally across the screen, the map seems to move vertically and not horizontally.

Reply:
I think what was happening here is that the app was often times picking up the `panStarted` and `panEnded` cases, while skipping over `panMoved`. This resulted in the small vertical bump seen when trying to pan. I removed the `panStarted` and `panEnded` cases which seems to fix the issue from occurring. However, it does appear that the `manager:didReceivePanningFrom fromPoint:toPoint:` can be unreliable in picking up pan movements.

Reply:
Shouldn't the result of the pan be in the opposite direction of the finger drag? My understanding is that it should move away from where your finger ends up.